### PR TITLE
Document CuPy memory pool sharing

### LINKED ISSUE #6018: Document shared memory pool
The memory pool sharing feature between ChainerX and CuPy is not documented (introduced by https://github.com/chainer/chainer/pull/5821). It is an experimental feature but should still be.

### PR DESCRIPTION
**Pull request by hvy**
Fixes #6018 .

Should be merged after the actual feature ~https://github.com/chainer/chainer/pull/5821~.

### CODE REVIEW

File: docs/source/chainerx/tips.rst [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Tips and FAQs
+=============
+
+GPU memory consuption is too high when used with Chainer with CuPy
```
Comment (niboshi) [inline docs/source/chainerx/tips.rst]: Typo: consuption  -> consumption

Comment (hvy) [reply docs/source/chainerx/tips.rst]: Ops, thanks. Fixed.

**Comment by niboshi** [comment]
Jenkins, test this please

File: docs/source/chainerx/tips.rst [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Tips and FAQs
+=============
+
+GPU memory consumption is too high when used with Chainer with CuPy
```
Comment (niboshi) [inline docs/source/chainerx/tips.rst]: I think "Chainer with " can be omitted?

Comment (hvy) [reply docs/source/chainerx/tips.rst]: Hm, yes indeed. Thanks. Fixed.

**Comment by niboshi** [comment]
Jenkins, test this please

**Pull request merged**